test/e2e/customconfigs/waitforpodsready_test.go

```diff
@@ -0,0 +1,340 @@
+			cfg := originalCfg.DeepCopy()
+			cfg.WaitForPodsReady = &configapi.WaitForPodsReady{
+				Enable:          true,
+				BlockAdmission:  ptr.To(true),
+				Timeout:         &metav1.Duration{Duration: util.TinyTimeout},
+				RecoveryTimeout: nil,
+				RequeuingStrategy: &configapi.RequeuingStrategy{
+					Timestamp:          ptr.To(configapi.EvictionTimestamp),
+					BackoffBaseSeconds: ptr.To(int32(10)),
+					BackoffLimitCount:  ptr.To(int32(1)),
+				},
+			}
+
+			util.ApplyKueueConfiguration(ctx, k8sClient, cfg)
+			util.RestartKueueController(ctx, k8sClient)
+		})
+
+		ginkgo.By("creating a job that will never report its pods as ready", func() {
+			job = testingjob.MakeJob("job-timeout", ns.Name).
+				Queue(lq.Name).
+				Request(corev1.ResourceCPU, "2").
+				Parallelism(1).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+		})
+
+		wlKey = types.NamespacedName{
+			Name:      workloadjob.GetWorkloadNameForJob(job.Name, job.UID),
+			Namespace: ns.Name,
+		}
+
+		workload := &kueue.Workload{}
+		gomega.Eventually(func(g gomega.Gomega) {
+			g.Expect(k8sClient.Get(ctx, wlKey, workload)).Should(gomega.Succeed())
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+		ginkgo.By("waiting for the workload to be evicted", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlKey, workload)).Should(gomega.Succeed())
+
+				cond := apimeta.FindStatusCondition(workload.Status.Conditions, kueue.WorkloadPodsReady)
+				g.Expect(cond).ShouldNot(gomega.BeNil())
+				g.Expect(cond.Status).To(gomega.Equal(metav1.ConditionFalse))
+				g.Expect(cond.Reason).To(gomega.Equal(kueue.WorkloadWaitForStart))
+
+				evictedCond := apimeta.FindStatusCondition(workload.Status.Conditions, kueue.WorkloadEvicted)
+				g.Expect(evictedCond).ShouldNot(gomega.BeNil())
+				g.Expect(evictedCond.Status).To(gomega.Equal(metav1.ConditionTrue))
+				g.Expect(evictedCond.Reason).To(gomega.Equal(kueue.WorkloadEvictedByPodsReadyTimeout))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("verifying that the job is suspended", func() {
+			updatedJob := &batchv1.Job{}
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: job.Name, Namespace: ns.Name}, updatedJob)).Should(gomega.Succeed())
+				g.Expect(ptr.Deref(updatedJob.Spec.Suspend, false)).To(gomega.BeTrue())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("verifying that the workload is requeued", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				var wl kueue.Workload
+				g.Expect(k8sClient.Get(ctx, wlKey, &wl)).Should(gomega.Succeed())
+				g.Expect(wl.Status.RequeueState).ShouldNot(gomega.BeNil())
+				g.Expect(*wl.Status.RequeueState.Count).Should(gomega.Equal(int32(1)))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
```

Comment:
Can we also check if the workload is deactivated after the second eviction?

Reply:
Have added such a case